Tools/CMake/AMReXOptions.cmake

```diff
@@ -283,6 +283,8 @@ endif ()
 
 print_option( AMReX_ASSERTIONS )
 
+option(AMReX_BOUND_CHECK  "Enable bound checking in Array4 class" OFF)
```

Comment:
Oh cool, should we enable this by default with Debug builds?
How is the corresponding GNUmake logic?

Reply:
It's enabled with debug builds.

Reply:
My message was probably not very clear.  The `AMREX_BOUND_CHECK` macro is only used like below
```
#if defined(AMREX_DEBUG) || defined(AMREX_BOUND_CHECK)
```

Reply:
Ok perfect, then we can keep the build system logic simple in both GNUmake & CMake.